### DISCUSSION TITLE
Please revert to left and right vs start and end to maintain consistency

### DISCUSSION THREAD

**Discussion by mmestas** [twbs/bootstrap#34065] [General] [answered]
I've been trying out Bootstrap 5 and don't like how ml-2 has become ms-2, for example. 
It's inconsistent with css styling.  Everywhere, it's margin-left, border-left, padding-left, and bootstrap has stayed consistent with this until now. i.e. ml-2, pr-2, etc. Why the unnecessary change?!

**Comment by alpadev** [comment, marked as answer]
AFAIK, because we aim to support RTL (right-to-left text direction) languages. This swaps the directions between left <> right.

As such `margin-left` would add a margin to the right hand side, when the direction is switched to `rtl`. Since this changes the semantics, it was decided to replace them with start and end. 

We adapted basically to the transition that CSS went through in the past few years. There is for example `margin-inline-start` now, that translates into `margin-left` in a LTR environment and to `margin-right` in RTL.

**Reply by mmestas** [reply]
Thank you for that explanation

**Comment by markduncanawp** [comment]
Just to tag onto this discussion.  I would like to reinstate margin right, margin left, padding left and padding right for a migration period, so that i can switch to bootstrap 5 now and then go throught the thousands of webpages we have and convert them to ltr/rtl.  Is there a preferred approach, i assume i would need to add to my bootstrap customization/variables.  I can see the _utilities.scss which i understand i should not change so an example on how i should do this would be great.  Note the longer i leave this the more web page are created and therefore more work to do with the migration.
